simplq/src/components/pages/Join/Form.jsx

```diff
@@ -68,14 +68,14 @@ export function JoinQueueForm(props) {
     });
   };
 
-  const ActionButton = () => {
+  const CreateTokenButton = () => {
     if (addingInProgress)
       return (
         <div>
           <LoadingIndicator />
         </div>
       );
-    if (props.page === 'admin') return <AddToQueueButton onClick={handleClick} />;
+    if (props.buttonName === 'Add') return <AddToQueueButton onClick={handleClick} />;
```

Comment:
We can pass buttonName down to `CreateTokenButton`, then we won't need two separate `JoinQueueButton` and `AddToQueueButton `

Reply:
Done